be1-go/internal/handler/message/message_test.go

```diff
@@ -0,0 +1,311 @@
+
+	args = append(args, input{
+		name:     "Test 11",
+		message:  msg,
+		isError:  true,
+		contains: "failed to decode data",
+	})
+
+	// Test 12: success to handled message for channel root
+
+	channelPath = "rootMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(RootType, nil)
+
+	args = append(args, input{
+		name:        "Test 12",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 13: success to handled message for channel lao
+
+	channelPath = "laoMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(LaoType, nil)
+
+	args = append(args, input{
+		name:        "Test 13",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 14: success to handled message for channel election
+
+	channelPath = "electionMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(LaoType, nil)
+
+	args = append(args, input{
+		name:        "Test 14",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 15: success to handled message for channel chirp
+
+	channelPath = "chirpMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(ChirpType, nil)
+
+	args = append(args, input{
+		name:        "Test 15",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 16: success to handled message for channel reaction
+
+	channelPath = "reaction"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(ReactionType, nil)
+
+	args = append(args, input{
+		name:        "Test 16",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 17: success to handled message for channel coin
+
+	channelPath = "coinMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(CoinType, nil)
+
+	args = append(args, input{
+		name:        "Test 17",
+		channelPath: channelPath,
+		message:     msg,
+		isError:     false,
+	})
+
+	// Test 18: success to handled message for channel coin
+
+	channelPath = "coinMsg"
+	msg = generator.NewChirpAddMsg(t, sender, private, time.Now().Unix())
+
+	db.On("HasMessage", msg.MessageID).Return(false, nil)
+	db.On("GetChannelType", channelPath).Return(FederationType, nil)
+
+	args = append(args, input{
+		name:        "Test 18",
```

Comment:
Please rename the tests so that the name represents what the test is actually for. For someone running many tests at once, that's much more helpful than enumerating.

Reply:
We include comments describing each test before the test itself. Since we have the same structure for most test files, it should be easy to navigate and retrieve the comments. Do you think this is sufficient? I feel like it avoids having to write long names for each test.

Reply:
Nevermind, I didn't think about the problem of renaming all tests when we want to remove one since they are numbered.

Reply:
There are a lot of tests so it will be fixed in another pr.